### PR TITLE
Zcpy API: Remove unnecessary for loop from getSrcPe() method

### PR DESCRIPTION
**Pull request by nitbhat**

### CODE REVIEW

File: src/ck-core/ckrdma.C [hunk 1]
```diff
@@ -854,11 +854,10 @@ int getSrcPe(envelope *env) {
   PUP::fromMem up((void *)((CkMarshallMsg *)EnvToUsr(env))->msgBuf);
   up|numops;
   up|rootNode;
-  for(int i=0; i<numops; i++){
-    CkNcpyBuffer w;
-    up|w;
-    return w.pe;
-  }
+  CmiEnforce(numops > 0);
```
Comment (matthiasdiener) [inline src/ck-core/ckrdma.C]: Could be CmiAssert maybe

File: src/ck-core/ckrdma.C [hunk 2]
```diff
@@ -854,11 +854,10 @@ int getSrcPe(envelope *env) {
   PUP::fromMem up((void *)((CkMarshallMsg *)EnvToUsr(env))->msgBuf);
   up|numops;
```
Comment (stwhite91) [inline src/ck-core/ckrdma.C]: PUPing numops is useless here and below in getRootNode(). That can be done here or in a separate patch

Comment (matthiasdiener) [reply src/ck-core/ckrdma.C]: Isn't the goal of this just to unpack in the same way it was packed before?

Comment (nitbhat) [reply src/ck-core/ckrdma.C]: Unpacking numops and rootNode is required since that was how it was packed.

File: src/ck-core/ckrdma.C (hunk 1 above)
Comment (nitbhat) [reply src/ck-core/ckrdma.C]: Doesn't CmiAssert do nothing except for when built with '--enable-error-checking'?

Comment (matthiasdiener) [reply src/ck-core/ckrdma.C]: > Doesn't CmiAssert do nothing except for when built with '--enable-error-checking'?

Yes, I think so. I was just wondering if we should check the value of numops here in all build configs. If it's better with the check enabled, that's fine too.

**Comment by nitbhat** [comment]
This commit has been cherry-picked on top of charm-6.10.0

**Pull request merged**